File: documentation-examples-java/src/java/dev/typr/foundations/docs/duckdb/EnumType.java
package dev.typr.foundations.docs.duckdb;

import dev.typr.foundations.DuckDbType;
import dev.typr.foundations.DuckDbTypes;

@SuppressWarnings("unused")
public class EnumType {
  // start
  // Define your Java enum
  public enum Status {
    PENDING,
    ACTIVE,
    COMPLETED
  }

  // Create DuckDbType — pass values(), no reflection
  DuckDbType<Status> statusType = DuckDbTypes.ofEnum("status", Status.values());
  // stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/duckdb/EnumType.scala
package dev.typr.foundations.docs.duckdb

import dev.typr.foundations.{DuckDbType, DuckDbTypes}

@SuppressWarnings(Array("unused"))
object EnumType:
  //start
  // Define your Scala 3 enum
  enum Status:
    case PENDING, ACTIVE, COMPLETED

  // Create a DuckDbType using bimap (Scala 3 enums don't extend java.lang.Enum)
  val statusType: DuckDbType[Status] = DuckDbTypes.varchar.bimap(
    Status.valueOf,
    _.toString
  )
  //stop
